apps/mull-api/src/app/media/media.service.ts

```diff
@@ -13,6 +13,18 @@ export class MediaService {
     private mediaRepository: Repository<Media>
   ) {}
 
+  async uploadFile(file: FileUpload): Promise<Media | Error> {
+    try {
+      await this.saveFile(file);
+      var media = await this.createMedia(file.mimetype);
+      this.updateFilename(file.filename, media.id, media.mediaType);
+    } catch (err) {
+      return new Error('Internal Server Error');
+    }
+
+    return media;
+  }
+
```

Comment:
very clean

Reply:
It was actually nice getting to look at that method and realize it could be simplified, made me more comfortable doing stuff in the backend.